Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/BoundsControl.cs

```diff
@@ -1024,24 +1023,38 @@ private void TransformTarget(HandleType transformType)
                 }
                 else if (transformType == HandleType.Scale)
                 {
-                    float initialDist = Vector3.Dot(initialGrabPoint - oppositeCorner, diagonalDir);
-                    float currentDist = Vector3.Dot(currentGrabPoint - oppositeCorner, diagonalDir);
-                    float scaleFactor = 1 + (currentDist - initialDist) / initialDist;
+                    Vector3 scaleFactor = Target.transform.localScale;
+                    if (ScaleHandlesConfig.ScaleBehavior == HandleScaleMode.Uniform)
+                    {
+                        float initialDist = Vector3.Dot(initialGrabPoint - oppositeCorner, diagonalDir);
+                        float currentDist = Vector3.Dot(currentGrabPoint - oppositeCorner, diagonalDir);
+                        float scaleFactorUniform = 1 + (currentDist - initialDist) / initialDist;
+                        scaleFactor = new Vector3(scaleFactorUniform, scaleFactorUniform, scaleFactorUniform);
+                    }
+                    else // non-uniform scaling
+                    {
+                        // get diff from center point of box
+                        Vector3 initialDist = (initialGrabPoint - oppositeCorner);
+                        Vector3 currentDist = (currentGrabPoint - oppositeCorner);
+                        Vector3 grabDiff = (currentDist - initialDist);
+                        scaleFactor = Vector3.one + grabDiff.Div(initialDist);
 
```

Comment:
Can the user achieve this kind of interaction with the changes?

![bounds_control_scale](https://user-images.githubusercontent.com/16922045/90260805-f69ed180-de43-11ea-934b-c31f0763442e.gif)

Would be nice if we could have the option of specific handles per axis. Usefull when content layout needs to be maintained by only scaling the content mask object per axis.

Reply:
you can do that if you combine the non uniform scaling with the constraint manager - however the scale constraints are atm fixed on the corners - if you would want them to appear on the sides you would have to create your own version of scalehandles

Reply:
https://github.com/microsoft/MixedRealityToolkit-Unity/issues/8302